diff --git a/headers.h b/headers.h
index 512c85b..f5298fb 100755
--- a/headers.h
+++ b/headers.h
@@ -101,9 +101,6 @@
 #ifdef HAVE_SYS_PARAM_H
 # include <sys/param.h>
 #endif
-#ifdef HAVE_SYS_PTRACE_H
-# include <sys/ptrace.h>
-#endif
 #ifdef HAVE_SYS_REG_H
 # include <sys/reg.h>
 #endif
@@ -133,7 +130,9 @@
 #ifdef HAVE_ASM_PTRACE_H
 # include <asm/ptrace.h>
 #endif
-#ifdef HAVE_LINUX_PTRACE_H
+#ifdef HAVE_SYS_PTRACE_H
+# include <sys/prctl.h>
+#elif HAVE_LINUX_PTRACE_H
 # include <linux/ptrace.h>
 #endif
 #undef FU_ia64_fpreg
