--- a/crossdev	2014-04-18 16:19:51.262346358 +0300
+++ b/crossdev	2014-04-18 16:19:37.404927938 +0300
@@ -831,7 +831,7 @@
 	echo "cross-${CTARGET}/${pkg} ${use}" >> ${output}
 }
 set_links() {
-	local cat=$1 pkg=$2 ovl=$3
+	local cat=$1 pkg=$2 ver=$3 ovl=$4
 	local s srcdir=${PORTDIR} d
 
 	d="${CROSSDEV_OVERLAY}"/cross-${CTARGET}/${pkg}
@@ -847,7 +847,7 @@
 		exit 1
 	fi
 	for s in ${ovl} ${SEARCH_OVERLAYS} ; do
-		if [[ -d ${s}/${cat}/${pkg} ]] ; then
+		if [[ -f ${s}/${cat}/${pkg}/${pkg}-${ver}.ebuild ]] ; then
 			srcdir=${s}
 			einfo "getting ${cat}/${pkg} from ${srcdir}"
 			break
@@ -921,7 +921,7 @@
 	set_use ${pkg} ${use}
 	set_use_force ${pkg} -multilib
 	set_use_mask ${pkg} -multilib
-	set_links ${cat} ${pkg} "${ovl}"
+	set_links ${cat} ${pkg} ${ver} "${ovl}"
 	set_env ${l} ${pkg} "${env}"
 }
 set_metadata() {
